src/java.base/share/classes/java/lang/foreign/AbstractLayout.java

```diff
@@ -144,25 +144,49 @@ public boolean isPadding() {
         return this instanceof PaddingLayout;
     }
 
+    /**
+     * {@return the hash code value for this layout}
+     */
     @Override
     public int hashCode() {
         return name.hashCode() << Long.hashCode(alignment);
     }
 
+    /**
```

Comment:
This, and other javadoc for Object methods has to be copied from `MemoryLayout` to here, or the javadoc rendering for the layout subclasses will just display the Object method javadoc.